modules/services/podman-linux/builds.nix

```
@@ -0,0 +1,173 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.podman;
+
+  podman-lib = import ./podman-lib.nix { inherit lib config; };
+
+  createQuadletSource = name: buildDef:
+    let
+      buildConfig = (podman-lib.deepMerge {
+        Build = {
+          AuthFile = buildDef.authFile;
+          Environment = buildDef.environment;
+          File = buildDef.file;
+          ImageTag = ([ "homemanager/${name}" ] ++ buildDef.tags);
+          Label = (buildDef.labels // { "nix.home-manager.managed" = true; });
+          PodmanArgs = buildDef.extraPodmanArgs;
+          SetWorkingDirectory = buildDef.workingDirectory;
+          TLSVerify = buildDef.tlsVerify;
+        };
+        Install = {
+          WantedBy = (if buildDef.autoStart then [
+            "default.target"
+            "multi-user.target"
+          ] else
+            [ ]);
+        };
+        Service = {
+          TimeoutStartSec = 300;
+          RemainAfterExit = "yes";
+        };
+        Unit = {
+          Description = (if (builtins.isString buildDef.description) then
+            buildDef.description
+          else
+            "Service for build ${name}");
```

The parentheses are unnecessary

```suggestion
          Description = if isString buildDef.description then
            buildDef.description
          else
            "Service for build ${name}";
```

but I think the better solution is to make `description` have type `str` and default the value to `"Service for build ${name}"`. See my other comment.

---

I went ahead and made the changes in your other comment, then set this as `Description = buildDef.description;`. If it is null, then it is automatically dropped. Much simpler!